packages/jest-parallel/README.md

```diff
@@ -0,0 +1,107 @@
+# jest-parallel
+
+Module for executing heavy tasks under forked processes in parallel, by providing a modern interface (`Promise` based), minimum overhead, and sticky workers.
+
+The module works by providing an absolute path of the module to be loaded in all forked processes, as well as the list of methods that can be remotely called. All methods are exposed on the parent process as promises, so they can be `await`'ed. Child (worker) methods can either be synchronous or asynchronous.
+
+The way sticky workers work is by using the returned string of the `getWorker` method. If the string was used before, the call will be queued to the related worker; if not, it will be executed by the first available worker, then sticked to the worker that executed it; so the next time it will be processed by the same worker. If you have no preference on the worker executing the task, but you have defined a `getWorker` method because you want _some_ of the tasks to be sticked, you can return `null` from it.
+
+
+## Install
+
+```sh
+$ npm install --save jest-parallel
+```
+
+
+## API
+
+The only exposed method is a constructor (`JestFarm`) that is initialized by passing an options object. This object can have the following properties:
+
+
+### `exposedMethods: $ReadOnlyArray<string>` (required)
```

Comment:
What if you want to call a `module.exports = function() {}`? This is done internally in jest now

I came here to ask about using the babel interop to make transpiled `export default` work, but if a default export is not supported, I guess we can just do `exposedMethods: ['default']`

Reply:
Yep, the way `default` exports are exported by Babel is by adding a `default` key to the object, so it is technically possible to call them with the current implementation.

That said, standard exports can't be called, so this is a fair point. I will change the default exported object of `getApi` from an `Object.create(null)` to a bound `_makeCall` that calls with `default`.

Reply:
Example of where jest needs the default export (can be changed of course):

https://github.com/facebook/jest/blob/95278b2047051244f6cce416c1fbdebcd807265e/packages/jest-haste-map/src/worker.js#L43-L44

Reply:
I modified the code, so `api` is now also a wrapped call :)

Reply:
Awesome!

Any chance of using the babel interop instead of plain `require`, or do you think that'd be confusing behavior?

Reply:
And to clarify, is the way you invoke the main function by passing `exposedMethods: ['']` or `exposedMethods: []`?

Reply:
Yes, I'm fine implementing the Babel logic to be able to treat Babel-ified modules as if they were a standard one. Regarding the `['']`, it is not necessary to provide it, the default export is always exposed (i.e. the result of `getApi()` is always a method). This makes me think that `exposedMethods` should then become optional (there is no sense in making `[]` a required option).

I will also adjust `README.md` to add more information about default exports and potentially provide an example on how to use it.